shortcodes/ucf-charts-shortcode.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Handles registering the ucf-chart shortcode
+ **/
+if ( ! class_exists( 'UCF_Chart_Shortcode' ) ) {
+	class UCF_Chart_Shortcode {
+		/**
+		 * Registers the ucf-chart shortcode
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 **/
+		public static function register() {
+			add_shortcode( 'ucf-chart', array( 'UCF_Chart_Shortcode', 'handler' ) );
+		}
+
+		/**
+		 * The handler for the ucf-chart shortcode
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 * @param array $atts The array of shortcode attributes
+		 * @return string The shortcode markup
+		 **/
+		public static function handler( $atts ) {
+			$atts = shortcode_atts(
+				array(
+					'id'    => null,
+					'class' => ''
+				),
+				$atts
+			);
+
+			$id    = $atts['id'];
+			$class = $atts['class'];
+
+			if ( $id ) {
+				$chart = get_post( $id );
+
+				if ( 'ucf_chart' !== $chart->post_type ) return '';
+
+				$name         = $chart->post_name;
+				$class        = ! empty( $class ) ? 'custom-chart ' . $class : 'custom-chart';
+				$chart_type   = get_post_meta( $chart->ID, 'ucf_chart_type', TRUE );
```

Comment:
Does the chartjs lib assume a default chart_type if one isn't set by the user? (Do we need to enforce a default here?)

Reply:
I suppose it's possible for a Chart CPT to be saved without a chart type being selected, but it would require the user to manipulate the DOM. There is no "null" or "blank" value in the form, so a chart type will always be selected when the post is saved. In addition, the default behavior of the javascript that initiates the chart is to display nothing if there is no chart type.